core/src/components/select/select.tsx

```diff
@@ -740,14 +743,140 @@ export class Select implements ComponentInterface {
     );
   }
 
+  componentDidRender() {
+    /**
+     * Run this the frame after
+     * the browser has re-painted the select.
+     * Otherwise, the label element may have a width
+     * of 0 and the IntersectionObserver will be used.
+     */
+    raf(() => {
```

Comment:
Are there any performance considerations and/or optimizations we should handle here?

`componentDidRender` will call after each `render()` of the select, which can occur many times over the lifespan of the `ion-select` in the view. Each time that occurs we are invoking a `requestAnimationFrame` callback. 

Maybe a simpler way to ask this question - is there any conditions where we can opt out of calling  the `raf`/`setNotchWidth`? Such as if they are using legacy syntax, not using a visible label, etc. I understand the limitations with accessing the width in the current frame.

Reply:
`setNotchWidth` returns early if neither the label prop nor the label slot is used. This causes `setNotchWidth` to not run on legacy selects and selects without a visible label. It also returns early if `this.label` is defined, causing it to only run on selects with a label slot.

We potentially need to re-check the slot width if the `labelPlacement` or `label` props change or if the slotted content changes. For example, if the `label` prop becomes defined then we want to remove the explicit `width` CSS property. Additionally, if the `labelPlacement` prop changes from `start` to `stacked` then we need to compute the notch width.

Reply:
Oh I guess I could move the `raf` into `setNotchWidth` so it is subject to the same early return checks. Is that what you are referring to?

Reply:
Yes, I was curious if the checks inside `setNotchWidth` are dependent on being called within `requestAnimationFrame` or can we opt to do the `raf` only after the early returns?

Reply:
Good call. I optimized this in https://github.com/ionic-team/ionic-framework/pull/27483/commits/714f8c470505360fdbbefe6c2611aa6fef6e1e82.

(I removed the extra console.log in https://github.com/ionic-team/ionic-framework/pull/27483/commits/2339727c8e8871e5074a8036fe9265be505e75ef)